packages/amplication-client/src/Application/appSettings/DirectoriesSettingsForm.tsx

```diff
@@ -0,0 +1,208 @@
+            validate(values, FORM_SCHEMA)
+          }
+          enableReinitialize
+          onSubmit={handleSubmit}
+        >
+          {(formik) => {
+            return (
+              <Form>
+                <div className={`${CLASS_NAME}__header`}>
+                  <h3>Base directories</h3>
+                </div>
+                <p className={`${CLASS_NAME}__description`}>
+                  Amplication gives you the choice of which components to
+                  generate. Use the settings to include or exclude GraphQL API,
+                  REST API, and Admin UI.
+                </p>
+                <hr />
+                <FormikAutoSave debounceMS={1000} />
+                <Panel panelStyle={EnumPanelStyle.Transparent}>
+                  <h2>Server</h2>
+                  <TextField
+                    className={`${CLASS_NAME}__formWrapper_field`}
+                    name="serverSettings[serverPath]"
+                    placeholder="./packages/[SERVICE-NAME]"
+                    label="Server base URL"
+                    value={data?.appSettings.serverSettings.serverPath || ""}
+                    helpText={data?.appSettings.serverSettings.serverPath}
+                    labelType="normal"
+                  />
+                </Panel>
+                <hr />
+                <Panel panelStyle={EnumPanelStyle.Transparent}>
+                  <h2>Admin UI</h2>
+                  <TextField
+                    className={`${CLASS_NAME}__formWrapper_field`}
+                    name="adminUISettings[adminUIPath]"
+                    placeholder="./packages/[SERVICE-NAME]"
+                    label="Admin UI base URL"
+                    disabled={!data?.appSettings.serverSettings.generateGraphQL}
+                    value={data?.appSettings.adminUISettings.adminUIPath || ""}
+                    helpText={data?.appSettings.adminUISettings.adminUIPath}
+                    labelType="normal"
+                  />
+                </Panel>
+              </Form>
+            );
+          }}
+        </Formik>
+      )}
+      <Snackbar
+        open={Boolean(error)}
+        message={formatError(error || updateError)}
+      />
+    </div>
+  );
+}
+
+export default GenerationSettingsForm;
+
+const UPDATE_APP_SETTINGS = gql`
+  mutation updateAppSettings($data: AppSettingsUpdateInput!, $appId: String!) {
+    updateAppSettings(data: $data, where: { id: $appId }) {
+      id
+      dbHost
+      dbName
+      dbUser
+      dbPassword
+      dbPort
+      authProvider
+      serverSettings {
+        generateGraphQL
+        generateRestApi
+        serverPath
+      }
+      adminUISettings {
+        generateAdminUI
+        adminUIPath
+      }
+    }
+  }
+`;
+
+const GET_APP_SETTINGS = gql`
+  query appSettings($id: String!) {
+    appSettings(where: { id: $id }) {
+      id
+      dbHost
+      dbName
+      dbUser
+      dbPassword
+      dbPort
+      authProvider
+      serverSettings {
```

Comment:
import the queries from the one of the other forms instead of declaring it again